### DISCUSSION TITLE
[V5] How to validate nullable field?

### DISCUSSION THREAD

**Discussion by woubuc** [adonisjs/core#1978] [Help]
Validator schema:
```ts
name: schema.string.optional({ trim: true }),
description: schema.string.optional(),
```

Route handler in controller:
```ts
let body = await request.validate(Validator);

let item = await Item.findOrFail(params.id);
item.merge(body);
await item.save();
```

Now whenever I send `{ description: null }` or `{ description: "" }` (any falsy value) to the update request, the validator strips that field entirely. So it doesn't set the `description` field to `null` even though the column is nullable and I explicitly sent `null` in the request.

How can I make the validator return this field as null when it's set to a null value? Is there something like `schema.string.nullable()`?

**Comment by thetutlage** [comment]
Inside `config/app.ts` file under the `validator` object, you can define the option `existsStrict = true` to allow empty strings. Do note, the `null` and `undefined` both are considered missing and ignored.

If the `validator` object inside the config file is missing, then you can declare it as follows

```ts
export const validator: ValidatorConfig = {
  existsStrict: true,
}
```

And then make sure you are running `@adonisjs/core@5.0.4-preview-rc-2.1`

**Reply by woubuc** [reply]
> `null` and `undefined` both are considered missing and ignored

This is kind of the problem I'm having. I want `null` to mean 'set this nullable field to null (i.e. the value that I sent)' and not 'this field is missing so don't do anything'. That's what `undefined` is for imo.

The strict option gets me part of the way there (so I can workaround it) but not entirely.

**Reply by thetutlage** [reply]
Agree. I believe, we have to introduce another option to allow null values. I will look into it in couple of days and get back to you

**Reply by sn0rcha** [reply]
Any update on this @thetutlage?

**Reply by emiliosanches** [reply]
@thetutlage any update?

**Reply by lukfor85** [reply]
Any update please? I desperately need this tbh.

**Comment by sn0rcha** [comment]
I have the same problem. I need to pass null as a value when clearing the field but currently the validator strips it.

**Comment by mcordoba** [comment]
same problem here, any update?

**Comment by RomainLanz** [comment]
I am working on it this weekend. 👍🏻

**Reply by vaso2** [reply]
Supercool, waiting! Thanks!

**Reply by smit-code** [reply]
Waiting

**Reply by mcordoba** [reply]
waiting! thanks!

**Reply by loranbraga** [reply]
Any solution?

**Reply by dilenio** [reply]
Same problem here, waiting for a solution, ty!

**Reply by amadeujunior1101** [reply]
Waiting for a solution!

**Reply by thetutlage** [reply]
The support for nullable fields is added to `@adonisjs/core@5.4.2`. Here are the docs for the same https://docs.adonisjs.com/guides/validator/introduction#working-with-optional-and-null-values

**Reply by sn0rcha** [reply]
You rock, thanks! 🥇

**Reply by amadeujunior1101** [reply]
Great, thanks!

**Reply by dilenio** [reply]
ty so much

**Reply by loranbraga** [reply]
Hello @thetutlage, when I send "", I can get the value from request.all(), but when I get it with a validator there is no variable return. I can only get the attribute when I send null through it, in this case through json.

**Reply by sockenklaus** [reply]
> Inside `config/app.ts` file under the `validator` object, you can define the option `existsStrict = true` to allow empty strings. Do note, the `null` and `undefined` both are considered missing and ignored.
> 
> If the `validator` object inside the config file is missing, then you can declare it as follows
> 
> ```ts
> export const validator: ValidatorConfig = {
>   existsStrict: true,
> }
> ```
> 
> And then make sure you are running `@adonisjs/core@5.0.4-preview-rc-2.1`

@loranbraga 
Did you read this?

**Reply by rodrigorz** [reply]
To work with `nullableAndOptional` i have to disable empty strings `existsStrict: false` inside `config/app.ts`.
Because the `birth_date: schema.date.nullableAndOptional()` rule throws a `date.format` error when receive an empty string.

I am using Adonis as an API, all requests is a JSON or MULTIPART. Why `BodyParserJSONConfig` do not have `convertEmptyStringsToNull`? Only FORM / MULTIPART have it.

**Reply by thetutlage** [reply]
Because with JSON requests you can control what you are sending. Whereas, `convertEmptyStringsToNull` is meant to overcome the browsers behavior of sending empty string when a form field has no value.

If you are self constructing the JSON, then you can do this on the client.

**Reply by filev-svt** [reply]
I agree partially - I do have control over the JSON requests, however I would have to convert every single text input from an empty string to null manually (using react-hook-form), which would be far more difficult than just setting `convertEmptyStringsToNull` to true in the bodyparser.ts for JSON requests. I feel like it would be much more flexible to at least have the option to set this configuration.

**Reply by thetutlage** [reply]
We added it for v6. Lemme see if we can back port it for current version too
